ui/jobs/jobs.js

```diff
@@ -1827,24 +1710,24 @@ class Bars {
       }
     });
 
-    this.abilityFuncMap[kAbility.Biolysis] = () => {
+    this.abilityFuncMap[PveAction.SCH.Biolysis.HexID] = () => {
       bioBox.duration = 0;
       bioBox.duration = 30;
     };
-    this.abilityFuncMap[kAbility.Bio] = () => {
+    this.abilityFuncMap[PveAction.SCH.Bio.HexID] = () => {
       bioBox.duration = 0;
       bioBox.duration = 30;
     };
-    this.abilityFuncMap[kAbility.Bio2] = () => {
+    this.abilityFuncMap[PveAction.SCH.BioIi.HexID] = () => {
```

Comment:
Not to be discouraging, but I find this structure to be both confusing to read and full of a lot of information that's not needed.  Are the other fields valuable to have in JavaScript?  Is there something you're planning on doing with them that you need at runtime in an overlay vs in Python?

Personally, I would rather see output along the lines of https://github.com/quisquous/cactbot/blob/main/resources/effect_id.js, where there is a map of English name to a hex id.  I think you could use your script to only care about valid player pve abilities to make this an easier process.  If you wanted other tables, I think I could imagine generating `Ability` vs `CraftingAbility`?

What do you think of that approach?

Reply:
Not discouraging at all.  I know that I'm very new to open source and I welcome any feedback I can get to improve.

I definitely agree with trimming it down, I don't think all of the fields are relevant for the JS.  The main reason I included all of them was simply to show that they're there.  I'd rather have someone else agree they're not relevant than just decide on my own since I have somewhat limited experience in here.

I do think the effect_id.js format is a little too limiting.  There are a couple things I think would be useful in the JS, like recast, cost, preservescombo, maybe one or two others.

I also think it is necessary to include the class in some manner.  I can see it being action.CLS.name or action.name.CLS depending on what level of consistency we're looking for.  I personally prefer doing class.name because it is a more object oriented approach.  Also, always specifying the class makes it so that I don't need to remember which actions have multiple versions.

Reply:
I think if somebody wants to see what fields exists, they can go look at the data itself.  I don't feel like cactbot should include all of the fields just because they exist.

I continue to agree with the comments of others above that I don't think the class name needs to be included.  It feels like an edge case where there are a few conflicts, but for the most part pve ability names are unique among all classes and jobs.  I think it's a more object oriented approach, but I also find it less readable.  I have to do a lot more thinking in order to remember the name for something.